File: Java_Codes/array_binarySearch_iterative.java
import java.util.*;
public class array_binarySearch_iterative {

    int binarySearch(int array[], int length, int y, int num){
        while (y <= length) {
            int mid = y + (length-y)/2;
            if (array[mid]==num) {
                return mid;
            }

            if (array[mid]>num) {
                length = mid -1;
            }
            else{
                y = mid +1;
            }
        }
        return -1;
    }
    public static void main(String[] args){
        Scanner scr = new Scanner(System.in);
        array_binarySearch_iterative func = new array_binarySearch_iterative();
        int size, key;

        System.out.print("Enter the size of the array: ");
        size = scr.nextInt();
        
        int[] myArray = new int[size];
        System.out.println("Enter th elements of the array: ");
        for (int i = 0; i < myArray.length; i++) {
            myArray[i] = scr.nextInt();
        }

        System.out.print("The entered array is: ");
        for (int i = 0; i < myArray.length; i++) {
            System.out.print(myArray[i] + " ");
        }

        System.out.print("\nEnter the element to be searched in the array: ");
        key = scr.nextInt();

        int result = func.binarySearch(myArray, myArray.length-1, 0, key);
        if (result == -1) {
            System.out.println("The element is not in the array.");
        }
        else{
            System.out.println("The element is at "+result+"th indices of array.");
        }
        scr.close();
    }
}

File: C++_Codes/array_binarySearch_iterative.cpp
#include<iostream>
int binarySearch(int arr[], int length){
    int x;
    int y = 0;

    std::cout << "\nEnter the element to be searched in the array: ";
    std::cin >> x;

    while (y <= length)
    {
        int mid = y + (length - y) / 2;
        if (arr[mid] == x)
        {
            return mid;
        }
        if (arr[mid] < x)
        {
            y = mid + 1;
        }
        else
        {
            length = mid - 1;
        }
    }
    return -1;
}

int main(){
    int size;

    std::cout << "Enter the size of the array: ";
    std::cin >> size;

    int myArray[size];
    std::cout << "Enter the elements of the array: ";
    for (int i = 0; i < size; i++)
    {
        std::cin >> myArray[i];
    }
    std::cout << "The entered array is: ";
    for (int i = 0; i < size; i++)
    {
        std::cout << myArray[i] << " ";
    }
    int result = binarySearch(myArray, size-1);
    if (result == -1)
    {
        std::cout << "The element is not in the array.";
    }
    else
    {
        std::cout << "The element is at " << result << "th index of the array.";
    }
    return 0;
}
